### PR TITLE
ROC-5309: Store case name in ccd

### PR DESCRIPTION
**Pull request by karthikkannankainos**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-5309

### Change description ###
Store Claim name in the CCD.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/util/MapperUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    }
+
+    @Test
+    public void caseNameWhenMultiDefendant() {
+
+        Claim claimWithMultiDefendant = Claim.builder().claimData(
+            SampleClaimData.builder(singletonList(SampleParty.builder().withName("Brexiter").individual()),
+                Arrays.asList(
+                    SampleTheirDetails.builder().withName("Theresa May").individualDetails(),
+                    SampleTheirDetails.builder().withName("John Beckrow").individualDetails(),
```
Comment (andrewwa-kainos) [inline ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/util/MapperUtilTest.java]: Typo, `Bercow` 😁

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/util/MapperUtil.java [hunk 2]
```diff
@@ -1,19 +1,50 @@
...
+
+    private static String fetchDefendantName(Claim claim) {
+        StringBuilder defendantNameBuilder = new StringBuilder();
+
+        if (claim.getResponse().isPresent()) {
+            defendantNameBuilder
+                .append(claim.getResponse().map(response -> response.getDefendant().getName()).orElse(""));
+        } else {
+            defendantNameBuilder.append(claim.getClaimData().getDefendants().get(0).getName());
+        }
```
Comment (andrewwa-kainos) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/util/MapperUtil.java]: Assuming the `orElse` above is just there to satisfy the `Optional` (our business rules should guarantee a response's defendant's name is present) this whole block can be rewritten:
```
claim.getResponse()
    .map(Response::getDefendant)
    .map(Party::getName)
    .orElseGet(() -> claim.getClaimData().getDefendants().get(0).getName())
```

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/util/MapperUtil.java [hunk 3]
```diff
@@ -1,19 +1,50 @@
...
+
+    private static String fetchClaimantName(Claim claim) {
+        StringBuilder claimantNameBuilder = new StringBuilder();
+
+        claimantNameBuilder.append(claim.getClaimData().getClaimants().get(0).getName());
+        if (claim.getClaimData().getClaimants().size() > 1) {
+            claimantNameBuilder.append(OTHERS);
+        }
+        return claimantNameBuilder.toString();
+    }
```
Comment (andrewwa-kainos) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/util/MapperUtil.java]: Minor comment: for simple concatenations, `StringBuilder` is less performant than just `+`. Doesn't make a lot of difference for our low-volume solution though.

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/util/MapperUtil.java (hunk 2 above)
Comment (karthikkannankainos) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/util/MapperUtil.java]: Will try this, seems correct.

**Pull request merged**